contracts/acl/ACL.sol

```diff
@@ -421,7 +424,7 @@ contract ACL is IACL, TimeHelpers, AragonApp, ACLHelpers {
 
         // a raw call is required so we can return false if the call reverts, rather than reverting
         bytes memory checkCalldata = abi.encodeWithSelector(sig, _who, _where, _what, _how);
-        uint256 oracleCheckGas = ORACLE_CHECK_GAS;
+        uint256 oracleCheckGas = gasleft().sub(ERROR_GAS_ALLOWANCE);
```

Comment:
Wondering about the use of `SafeMath` here, since we don't use it in the [`DelegateProxy`](https://github.com/aragon/aragonOS/blob/next/contracts/common/DelegateProxy.sol#L20) when reserving gas from the delegatecall.

What does the EVM do if this underflows and we pass the extremely large, underflowed value to `staticcall`? Does it allow the called contract to eat up all the remaining gas?

Reply:
My tests on remix suggest it saves ~1/64th of the total gas available, as you mentioned elsewhere, even when passed `uint(-1)` (max number). In which case shall I remove `SafeMath`?

Reply:
Let's skip SafeMath, but error with the revert if the subtraction would underflow.

I'm thinking something like

```solidity
bool success = false;

if (gasleft() > ERROR_GAS_ALLOWANCE) {
  // do call with adjusted gas limit
}

if (!success) {
  if (gasleft() < ERROR_GAS_ALLOWANCE) {
    revert('ERROR');
  }
  return false;
}

Reply:
I'll remove SafeMath and can add the revert check. However, it might be worth noting that `gasleft()` will always be `> 50000` for transactions given over `~3200000` gas due to the 1/64th gas withholding. This means there could be times when the `staticcall` goes OOG but it doesn't reach the `revert()` statement and just returns `false`. The only alternative I can see would be to increase the `ERROR_GAS_ALLOWANCE` to `~125000` gas.

Reply:
It turns out that `hasPermission()` calls `evalParam()` twice so even with the lower `ERROR_GAS_ALLOWANCE` it does revert as expected since the second time round it only has the left over gas from the first time, which is tiny and will be smaller than the `ERROR_GAS_ALLOWANCE` once it reaches the revert. Pretty confusing though.

Reply:
> It turns out that hasPermission() calls evalParam() twice so even with the lower

Ahh yeah, this happens if you set it for `ANY_ENTITY`. However I think we can make a more concise test once we handle the 1/64 gas.

Good catch on the 1/64 stipend though!